### DISCUSSION TITLE
Force refresh of some languages

### DISCUSSION THREAD

**Discussion by Greedquest** [github-linguist/linguist#5595] [Classification] [answered]
Visual Basic support in linguist was split into VBA and VB.Net. This was in part to allow VBA repos to be found in github search. A comment on the [pull request](https://github.com/github/linguist/pull/4725) said:

> The rename should take effect in _some places_ when I make the next release, but only for "Visual Basic" -> "Visual Basic .NET" rename as the language ID is the same. Everything else will only take affect on those repos on the next push. The reason I say "some places" is because different parts of the site are cached differently and I have no idea how they all tie in.

The problem is, VBA is an old language and the majority of those repos are not actively maintained, so will never get the required push to clear the cache. An option would be for me to go and submit some tiny change to each one, however this is a chicken-and-egg problem because I can't find the VBA repos since they are all mistagged!

Is there any way to get this refresh to be run manually, on all VB.Net repos that have not been pushed to since the changes were made, and therefore may be VBA repos in disguise?

---

e.g. 
 - https://github.com/byronwall/bUTL
 - https://github.com/carvetighter/VBA-Code-Library
 - https://github.com/VBA-tools/VBA-Web
 - https://github.com/VBA-tools/VBA-JSON
 - 
All searchable because they happen to contain the word vba, but this repo https://github.com/AyrA/ExcelStuff contains loads of useful VBA, mistagged because it hasn't been pushed to, yet cannot be found by searching for the keyword "VBA".

**Comment by lildude** [comment, marked as answer]
> Is there any way to get this refresh to be run manually, on all VB.Net repos that have not been pushed to since the changes were made, and therefore may be VBA repos in disguise?

No, not easily. The functionality to do this easily doesn't exist so would need "sponsorship" for the feature to be planned and implemented. Unfortunately, this isn't likely to happen.
